src/components/table.tsx

```diff
@@ -37,30 +40,79 @@ function Row({
   );
 }
 
+function textContent(v: VNode): string {
+  const doc = document.implementation.createHTMLDocument("");
+  render(v, doc.body);
+  return doc.body.textContent || "";
+}
+
+interface CSVProps {
+  "data-csv": string[];
+  colSpan: number;
+}
+
+function toCSVValues(cell: TableCell): string[] {
+  if (typeof cell === "object") {
+    if ("data-csv" in cell.props) {
+      return (cell.props as unknown as CSVProps)["data-csv"];
+    }
+
+    const text = textContent(cell);
+    const colspan = (cell.props as unknown as CSVProps).colSpan || 1;
+
+    const empties = Array.from(Array(colspan - 1).keys()).map(() => "");
+
+    return [text, ...empties];
+  }
+  return [String(cell)];
+}
+
+function toCSV(data: TableData): string {
+  const { header, body } = data;
+
+  const rows = [
+    header.flatMap((v) => toCSVValues(v)),
+    ...body.map((row) => row.flatMap((v) => toCSVValues(v))),
+  ];
+
+  return rows.map((row) => row.map((c) => csvFormatValue(c)).join(",")).join("\n");
+}
+
 function Table({ data, numberFormatter = String }: TableProps): VNode {
   const { header, body, footer } = data;
   return (
-    <div className="usa-table-container--scrollable">
-      <table className="usa-table usa-table--compact">
-        <thead>
-          <tr>
-            {header.map((head) =>
-              typeof head === "object" && head.type === "th" ? head : <th>{head}</th>
-            )}
-          </tr>
-        </thead>
-        <tbody>
-          {body.map((row) => (
-            <Row row={row} numberFormatter={numberFormatter} />
-          ))}
-        </tbody>
-        {footer && (
-          <tfoot>
-            <Row row={footer} numberFormatter={numberFormatter} />
-          </tfoot>
-        )}
-      </table>
-    </div>
+    <>
+      <div className="usa-table-container--scrollable">
+        <table className="usa-table usa-table--compact">
+          <thead>
+            <tr>
+              {header.map((head) =>
+                typeof head === "object" && head.type === "th" ? head : <th>{head}</th>
+              )}
+            </tr>
+          </thead>
+          <tbody>
+            {body.map((row) => (
+              <Row row={row} numberFormatter={numberFormatter} />
+            ))}
+          </tbody>
+          {footer && (
+            <tfoot>
+              <Row row={footer} numberFormatter={numberFormatter} />
+            </tfoot>
+          )}
+        </table>
+      </div>
+
+      <a
+        className="usa-button usa-button--outline"
+        download="report.csv"
+        href={`data:text/csv;charset=utf-8,${encodeURIComponent(toCSV(data))}`}
```

Comment:
for most of our pages, the tables are not too big, but this is adding a big-ish string and just letting it hang out in the DOM. maybe if this becomes slow we could make this an async operation